src_c/rwobject.c

```diff
@@ -722,22 +737,43 @@ _rwops_from_pystr(PyObject *obj)
     if (obj != NULL) {
         SDL_RWops *rw = NULL;
         PyObject *oencoded;
+        char* encoded = NULL;
+        char* ext = NULL;
+        char* extension = NULL;
+
+        //longest extension is 4 chars, plus NULL at end
+        extension = malloc(sizeof(char)*5);
+        if (extension == NULL) {
+            return (SDL_RWops*)PyErr_NoMemory();
+        }
+
         oencoded = pg_EncodeString(obj, "UTF-8", NULL, NULL);
         if (oencoded == NULL) {
             return NULL;
         }
         if (oencoded != Py_None) {
-            rw = SDL_RWFromFile(Bytes_AS_STRING(oencoded), "rb");
+            encoded = Bytes_AS_STRING(oencoded);
+            rw = SDL_RWFromFile(encoded, "rb");
+            ext = strrchr(encoded, '.');
+            if (ext) {
+                ext++;
+                strcpy(extension, ext);
+            }
         }
         Py_DECREF(oencoded);
         if (rw) {
+            /* adding the extension to the hidden data for RWops from files */
```

Comment:
+1 for great comment

Reply:
Thanks! I thought that this system would be very confusing to figure out for anyone new coming to look at the code.